wadsrc/static/zscript/engine/base.zs

```diff
@@ -140,6 +140,31 @@ enum EPrintLevel
 	PRINT_NOLOG = 2048,		// Flag - do not print to log file
 };
 
+/*
+// These are here to document the intrinsic methods and fields available on
+// the built-in ZScript types
+class Vector2
+{
+	double x;
+	double y;
+	native double Length();
+	native Vector2 Unit();
+}
+
+class Vector3 : Vector2
+{
+	double z;
+	Vector2 xy;
+	native double Length();
+	native Vector3 Unit();
+}
```

Comment:
These are structs, not classes. Otherwise you'd be able to put them in dynamic arrays (and no, you cannot - not directly).

Reply:
Additionally, it'd be worth mentioning the dot and cross products [as seen here.](https://zdoom.org/wiki/ZScript_special_words)

Reply:
> Additionally, it'd be worth mentioning the dot and cross products [as seen here.](https://zdoom.org/wiki/ZScript_special_words)

Which one looks better?

    // native double operator dot(Vector2 other); // Dot product

or:

    // The dot product of two vectors can be calculated like this:
    // double d = a dot b;

Reply:
The latter. Also, don't forget to include the x and y variable in vector3.

`double x, y, z;` will suffice.